File: javascript/src/models/googleCloudStoragePatchConfig.ts
// this file is @generated

export interface GoogleCloudStoragePatchConfig {
  bucket?: string;
  credentials?: string;
}

export const GoogleCloudStoragePatchConfigSerializer = {
  _fromJsonObject(object: any): GoogleCloudStoragePatchConfig {
    return {
      bucket: object["bucket"],
      credentials: object["credentials"],
    };
  },

  _toJsonObject(self: GoogleCloudStoragePatchConfig): any {
    return {
      bucket: self.bucket,
      credentials: self.credentials,
    };
  },
};

File: rust/src/models/google_cloud_storage_patch_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct GoogleCloudStoragePatchConfig {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub bucket: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub credentials: Option<String>,
}

impl GoogleCloudStoragePatchConfig {
    pub fn new() -> Self {
        Self {
            bucket: None,
            credentials: None,
        }
    }
}
